Mage.Sets/src/mage/cards/j/JeskaiInfiltrator.java

```diff
@@ -81,38 +70,31 @@ public JeskaiInfiltratorEffect copy() {
-        // TODO: migrade to shared manifest code
         Player controller = game.getPlayer(source.getControllerId());
-        if (controller != null) {
-            Set<Card> cardsToManifest = new HashSet<>();
-            cardsToManifest.add(source.getSourcePermanentIfItStillExists(game));
-            cardsToManifest.add(controller.getLibrary().getFromTop(game));
-            UUID exileId = UUID.randomUUID();
-            controller.moveCardsToExile(cardsToManifest, source, game, false, exileId, "");
-            ExileZone exileZone = game.getExile().getExileZone(exileId);
-            for (Card card : exileZone.getCards(game)) {
-                card.setFaceDown(true, game);
-            }
-            game.fireUpdatePlayersEvent(); // removes Jeskai Infiltrator from Battlefield, so Jeskai Infiltrator returns as a fresh permanent to the battlefield with new position
-
-            Ability newSource = source.copy();
-            newSource.setWorksFaceDown(true);
-            //the Set will mimic the Shuffling
-            exileZone.getCards(game).forEach(card -> {
-                ManaCosts manaCosts = null;
-                if (card.isCreature(game)) {
-                    manaCosts = card.getSpellAbility() != null ? card.getSpellAbility().getManaCosts() : null;
-                    if (manaCosts == null) {
-                        manaCosts = new ManaCostsImpl<>("{0}");
-                    }
-                }
-                Card battlefieldCard = BecomesFaceDownCreatureEffect.findDefaultCardSideForFaceDown(game, card);
-                MageObjectReference objectReference = new MageObjectReference(battlefieldCard.getId(), battlefieldCard.getZoneChangeCounter(game) + 1, game);
-                game.addEffect(new BecomesFaceDownCreatureEffect(manaCosts, objectReference, Duration.Custom, FaceDownType.MANIFESTED), newSource);
-            });
-            controller.moveCards(exileZone.getCards(game), Zone.BATTLEFIELD, source, game, false, true, false, null);
-            return true;
+        if (controller == null) {
+            return false;
+        }
+        UUID exileId = UUID.randomUUID();
+        Permanent sourcePermanent = source.getSourcePermanentIfItStillExists(game);
+        if (sourcePermanent != null) {
+            controller.moveCardsToExile(sourcePermanent, source, game, false, exileId, "");
+        }
+        Card topCard = controller.getLibrary().getFromTop(game);
+        if (topCard != null) {
+            controller.moveCardsToExile(topCard, source, game, false, exileId, "");
+        }
+        // need to get source permanent as card rather than permanent for next steps, hence this convoluted code
+        List<Card> cardsToManifest = new ArrayList<>(game.getExile().getExileZone(exileId).getCards(game));
+        if (cardsToManifest.isEmpty()) {
+            return false;
+        }
+        for (Card card : cardsToManifest) {
+            card.setFaceDown(true, game);
```

Comment:
doManifest also set face down status, but there are possible uses cases to keep it in exile zone. So that additional face down status setup must be correct. Need only additional comments about it in both cards like:
```
// need set face down before manifest:
// If an effect tries to return a face-down creature to the battlefield after it leaves (such as Aminatou’s second ability or Adarkar Valkyrie’s delayed triggered ability), that effect returns the card face up. If it tries to put an instant or sorcery card onto the battlefield this way, that card remains in its current zone instead. 
// (2018-07-13)
for 
setFaceDown
```

Reply:
Argh, nothing to do with it, it’s about face up. All face down allowed to move.